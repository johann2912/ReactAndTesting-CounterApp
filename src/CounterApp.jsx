import { useState } from 'react'
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);

    /**
     * add values for counter
     */
    const handleAdd = () => {
        setCounter(counter + 1);
    }

    /**
     * delete values for counter
     */
    const handleSubstract = () => setCounter(counter - 1);


    /**
     * reset count for counter 
     */
    const handleReset = () => setCounter(value);


    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>

            <button aria-label='btn-add' onClick={handleAdd} >
                +1
            </button>
            <button aria-label='btn-reset' onClick={handleReset} >
                Reset
            </button>
            <button aria-label='btn-substrac' onClick={handleSubstract} >
                -1
            </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}

CounterApp.defaultProps = {
    value: 'Does not found value'
}