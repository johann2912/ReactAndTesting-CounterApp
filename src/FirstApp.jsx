import PropTypes from 'prop-types';

export const FirstApp = ( { title, subTitle, name } ) => {

    

    return (
        <>
            <h1 data-testid="test-title" > { title } </h1>
            <p> { subTitle } </p>
            <p> { subTitle } </p>
            <p> { name } </p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
}

FirstApp.defaultProps = {
    name: 'Kakaroto',
    subTitle: 'Sub-title does not found',
    //title: 'Title does not found',
}