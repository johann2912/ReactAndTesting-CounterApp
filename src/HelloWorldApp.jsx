export const HelloWorldApp = () => {
    /**
     * document.createElement...
     */
    return (
        <h1>Hello world app</h1>
    );
};

//export default App;