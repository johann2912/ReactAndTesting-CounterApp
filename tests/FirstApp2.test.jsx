import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Test in <FirstApp />', () => {

    const title = 'Hola, Soy Goku';
    const subTitle = 'Soy un subtitulo';

    test('Should match snapshot', () => {
        const { container } = render(
            <FirstApp
                title={title}
            />
        )

        expect(container).toMatchSnapshot();
    });

    test('should show the message "Hola, Soy Goku"', () => {
        render(<FirstApp title={title} />)
        // screen.debug()
        expect(screen.getByText(title)).toBeTruthy();
    })

    test('should show title in an <h1>', () => {
        render(<FirstApp title={title} />)

        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title)
    })

    test('Should show the title sent by Props', () => {
        render(
            <FirstApp
                title={title}
                subTitle={subTitle}
            />
        )

        expect( screen.getAllByText(subTitle).length).toBe(2);
    })

});