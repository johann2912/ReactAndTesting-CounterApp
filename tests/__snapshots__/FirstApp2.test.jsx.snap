// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <FirstApp /> Should match snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, Soy Goku
     
  </h1>
  <p>
     
    Sub-title does not found
     
  </p>
  <p>
     
    Sub-title does not found
     
  </p>
  <p>
     
    Kakaroto
     
  </p>
</div>
`;
